src/Razor/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentPreserveWhitespaceDirective.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Language.Components
+{
+    internal static class ComponentPreserveWhitespaceDirective
+    {
+        public static readonly DirectiveDescriptor Directive = DirectiveDescriptor.CreateDirective(
+            "preservewhitespace",
+            DirectiveKind.SingleLine,
+            builder =>
+            {
+                builder.AddBooleanToken(ComponentResources.PreserveWhitespaceDirective_BooleanToken_Name, ComponentResources.PreserveWhitespaceDirective_BooleanToken_Description);
+                builder.Usage = DirectiveUsage.FileScopedMultipleOccurring;
```

Comment:
Any reason for this to be `FileScopedMultipleOccurring` instead of `FileScopedSinglyOccurring`?

Reply:
I thought the multiple occurring was needed for overriding, but if you’re saying that will still work with single occurring I’ll change it.

Reply:
From my understand it is to allow multiple directives in the same file. Singly occurring should still allow overriding what was set in _Imports.razor. @NTaylorMullen can you confirm?